c7n/resources/sqs.py

```diff
@@ -118,6 +119,31 @@ class SQSCrossAccount(CrossAccountAccessFilter):
     permissions = ('sqs:GetQueueAttributes',)
 
 
+@SQS.filter_registry.register('kms-alias')
+class KmsKeyAlias(ResourceKmsKeyAlias):
+    """
+    Filter a resource by its associated kms key aliasname 'AliasName'
+
+    :example:
+
+         .. code-block:: yaml
+
+             policies:
+                - name: sqs-kms-key-filters
+                  resource: sqs
+                  filters:
+                    - type: kms-alias
+                      key: AliasName
+                      value: "^(alias/aws/)"
+                      op: regex
+    """
+    def process(self, resources, event=None):
+        for resource in resources:
+            if 'KmsMasterKeyId' in resource:
```

Comment:
why the reassignment?

Reply:
I don't remember why now, but at first I was under the impression that SQS needed to use the `kms-alias` type filter, similar to the existing checks in [ebs](https://github.com/cloud-custodian/cloud-custodian/blob/8b9c9527f151e565d3be109586f2ca0273d61695/c7n/resources/ebs.py#L620) or [rds](https://github.com/cloud-custodian/cloud-custodian/blob/8b9c9527f151e565d3be109586f2ca0273d61695/c7n/resources/rds.py#L314)

I've updated the code to use the `kms-key` type filter, which seems more capable.

SQS is a little bit different than the others, in that the KmsMasterKeyId seems to contain the alias name, if it exists, instead of always being the key arn.  So that's kind of odd.  I tried to note that in the example filter rule.

Reply:
that should be fine, kms apis are pretty orthogonal to taking both, its the service apis that are consuming kms that are a bit more picky typically.